.travis_do

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -e
+
+echo_red() { printf "\033[1;31m$*\033[m\n"; }
+
+if [ "$TRAVIS_OS_NAME" == "linux" ] 
```

Comment:
thinking about this a bit, maybe it's a good idea [if moving everything to script files] to have them functionally separate; 

i.e. how a `astyle.sh` script that does only astyle logic, another script that handles builds (let's say `build_projects.sh`)

a monolithic script becomes hard to follow as it grows over time ;

for `cppcheck` you could move it into a script as well, but maybe later ; it's only one line in travis, so not sure if it's worth having it into it's own script file